src/main/src/main/java/org/geoserver/catalog/impl/StyleInfoImpl.java

```diff
@@ -133,6 +133,10 @@ public Style getStyle() throws IOException {
 
     @Override
     public StyledLayerDescriptor getSLD() throws IOException {
+        // to avoid test NPEs
+        if (catalog == null || catalog.getResourcePool() == null) {
+            return null;
+        }
```

Comment:
Is this needed for tests? Like, why didn't we get a NPE before now? Layer group defined by styles use this method in other places in the code base.